src/API/Google/AdsCampaign.php

```diff
@@ -388,6 +405,46 @@ protected function convert_campaign( GoogleAdsRow $row ): array {
 		return $data;
 	}
 
+	/**
+	 * Combine converted campaigns data with campaign criterion results data
+	 *
+	 * @param array $campaigns Campaigns data returned from a query request and converted by convert_campaign function.
+	 *
+	 * @return array
+	 */
+	protected function combine_campaigns_and_campaign_criterion_results( array $campaigns ): array {
+		$query_conditions = [
+			[ 'campaign.id', array_keys( $campaigns ), 'IN' ],
+			// negative: Whether to target (false) or exclude (true) the criterion.
+			[ 'campaign_criterion.negative', 'false', '=' ],
+			[ 'campaign_criterion.status', 'REMOVED', '!=' ],
+		];
+
+		$campaign_criterion_results = $this->get_ads_query_results( new AdsCampaignCriterionQuery(), $query_conditions );
+
+		/** @var GoogleAdsRow $row */
+		foreach ( $campaign_criterion_results->iterateAllElements() as $row ) {
+			$campaign           = $row->getCampaign();
+			$campaign_id        = $campaign->getId();
+			$campaign_criterion = $row->getCampaignCriterion();
+			$location           = $campaign_criterion->getLocation();
+
+			// TODO: Convert the geo_traget_constant to the ISO 3166-1 alpha-2 country code
+			// after https://github.com/woocommerce/google-listings-and-ads/issues/1229 is done.
+			$geo_target_constant = $location ? $location->getGeoTargetConstant() : null;
+
+			if ( ! array_key_exists( 'targeted_locations', $campaigns[ $campaign_id ] ) ) {
+				$campaigns[ $campaign_id ]['targeted_locations'] = [];
```

Comment:
The convert_campaign function already sets it to an empty array.

If I'd personally need to build in extra safety checks I'd rather skip the row if `$campaigns[ $campaign_id ]` is not set. It should never happen but it would seem more logical in case the query results are not what we expect.

Reply:
Agree, updated in f5bacf6.